### DISCUSSION TITLE
Replace files in tasmota folder

### DISCUSSION THREAD

**Discussion by ottelo9** [benzino77/tasmocompiler#266] [Q&A] [answered]
Hi there. I have a custom "xdrv_10_scripter.ino" that I want to include in the build. How can I make this?

**Comment by benzino77** [comment, marked as answer]
Generally TasmoCompiler is not build to compile sources other that Tasmota, but you can try to follow the steps described here:
https://github.com/benzino77/tasmocompiler/blob/master/ADVANCED.md

**Reply by ottelo9** [reply]
Thanks for your answer, but I mean I want to remove the file "xdrv_10_scripter.ino" from the tasmota folder with my modified.

**Reply by benzino77** [reply]
I'm not following ... :( You want to replace `xdrv_10_scripter.ino` with your own or you want to delete it completely?
TasmoCompiler assumes that sources of Tasmota are located in `/tmp/Tasmota`, so If you want manipulate sources you have to do this there. 
Remember that TasmoCompiler is switching git branches during its operations (switch from branch `develepmont` to `11.x.x` for example), so you have to make your changes in a branch which you will be using.

**Reply by ottelo9** [reply]
@benzino77 
Yes exactly. I want to replace the file, because I add some new code. I could use gitpod to compile tasmota with my modified file on my own, but this is little bit more work :).

Where can I find /tmp/Tasmota? I see now folder in gitpod.

**Reply by benzino77** [reply]
![image](https://user-images.githubusercontent.com/15658235/162173447-96d0b223-0fdd-48b0-8205-842409584218.png)

`/tmp/Tasmota` folder will be there after you click "Download Sources" in TasmoCompiler GUI.

**Reply by bepluy** [reply]
Could you please provide next steps after editing files in /tmp/Tasmota ?
I'm making all necessary changes in file (energy meter file), but after i compile - there's no changes.
Should i do something else ?

**Reply by benzino77** [reply]
Which version of TasmoCompiler are you using? Stable or Development?

**Comment by bepluy** [comment]
> Which version of TasmoCompiler are you using? Stable or Development?
Stable, simply following gitpod link.

**Reply by benzino77** [reply]
Sorry for the lag in response.

1. Do the first "dummy" compilation version of Tasmota  in which you want to introduce the changes - Tasmocompiler will switch to the appropriate branch under the hood.
2. Add `/tmp/Tasmota` folder to your gitpod workspace (as showed on the screenshot in this thread) 
3. Introduce your changes 
4. Commit them

It _should_ do the trick to "persist" your changes

All of that can be done via gitpod IDE.
